### PR TITLE
BuildResidentialHPXML measure: Support ambient foundations for SFA and Apartment units

### LINKED ISSUE #945: BuildResidentialHPXML measure: Support ambient foundations for SFA and Apartment units
As we do for SFD units.

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Closes #945.

Also:
* Cleans up `geometry.rb`
* Adds "extra" tests for SFA/Apartment units with ambient foundation
* Adds error check and test for Apartment unit with attic
* Relocates some `geometry.rb` error checks to `validate_arguments` method
* Adds missing tests for relocated error checks
* Fixes spelling in `workflow/template-build-hpxml-and-stocastic-schedules.osw` (=> `stochastic`)

## Checklist

Not all may apply:

- [ ] EPvalidator.xml has been updated
- [x] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: BuildResidentialHPXML/tests/build_residential_hpxml_test.rb [hunk 1]
```diff
@@ -187,28 +201,40 @@ def test_workflows
...
       'error-sfa-no-building-num-units.xml' => 'Did not specify the number of units in the building for single-family attached or apartment units.',
-      'error-sfa-above-apartment.xml' => 'Single-family attached buildings cannot be above another unit.',
-      'error-sfa-below-apartment.xml' => 'Single-family attached buildings cannot be below another unit.',
+      'error-sfa-above-apartment.xml' => 'Single-family attached units cannot be above another unit.',
+      'error-sfa-below-apartment.xml' => 'Single-family attached units cannot be below another unit.',
       'error-sfa-all-adiabatic-walls.xml' => 'At least one wall must be set to non-adiabatic.',
       'error-mf-no-building-num-units.xml' => 'Did not specify the number of units in the building for single-family attached or apartment units.',
       'error-mf-all-adiabatic-walls.xml' => 'At least one wall must be set to non-adiabatic.',
       'error-mf-two-stories.xml' => 'Apartment units can only have one above-grade floor.',
+      'error-mf-vented-attic.xml' => 'Apartment units can only have a flat roof or be below another apartment unit.',
```
Comment (shorowit) [inline BuildResidentialHPXML/tests/build_residential_hpxml_test.rb:217]: Wait, really? We don't allow a MF unit below an attic? Why is that and how much work would it be to accommodate it? I assume it's reasonably similar to handling a MF unit above a crawlspace or unconditioned basement.

File: docs/source/usage_instructions.rst [hunk 2]
```diff
@@ -58,7 +58,7 @@ A template OSW that simply runs the HPXMLtoOpenStudio, ReportSimulationOutput, a
 | This workflow automatically generates a CSV file with stochastic occupancy schedules before running the EnergyPlus simulation.
 
 | And another example:
-| ``openstudio run -w workflow/template-build-hpxml-and-stocastic-schedules.osw``
+| ``openstudio run -w workflow/template-build-hpxml-and-stochastic-schedules.osw``
```
Comment (shorowit) [inline docs/source/usage_instructions.rst:61]: Thanks for fixing this typo everywhere. That's embarrassing.

**Review by shorowit — APPROVED** [review]
This looks good. I didn't review the geometry.rb code changes, but I created two OSWs for an apartment unit, one above a crawlspace and one above an ambient foundation, and compared the two HPXML files. The latter correctly had no FoundationWalls, RimJoists, or Slab, and the FrameFloor's ExteriorAdjacentTo was changed to "outside". 💯 There are no other changes to HPXML files and all the other tests are happy, so that's good enough for me.

File: BuildResidentialHPXML/tests/build_residential_hpxml_test.rb (hunk 1 above)
Comment (joseph-robertson) [reply BuildResidentialHPXML/tests/build_residential_hpxml_test.rb:217]: Yes, we don't allow it. I believe this is an artifact of us historically only allowing flat roof for multifamily buildings. I opened a separate issue to support this: https://github.com/NREL/OpenStudio-HPXML/issues/965.

**Pull request merged**